src/test/java/com/github/wovnio/wovnjava/HeadersTest.java

```diff
@@ -333,8 +333,8 @@ public void testLocationWithSitePrefixPath() throws ConfigurationError {
         assertEquals("https://example.com/global/th/", h.locationWithLangCode("https://example.com/global/th/"));
         assertEquals("https://example.com/global/ja/tokyo/", h.locationWithLangCode("https://example.com/global/tokyo/"));
         assertEquals("https://example.com/global/ja/file.html", h.locationWithLangCode("https://example.com/global/file.html"));
-        assertEquals("https://example.com/global/ja/file.html", h.locationWithLangCode("https://example.com/pics/../global/file.html"));
-        assertEquals("https://example.com/global/../../file.html", h.locationWithLangCode("https://example.com/global/../../file.html"));
+        assertEquals("https://example.com/pics/../global/file.html", h.locationWithLangCode("https://example.com/pics/../global/file.html"));
```

Comment:
Is this OK to change expected result?

Reply:
This test (and the test below) asserts what happens if a servlet wants to redirect to a URL with explicit `/../` sections in its path.
```
language code to insert: "/ja/"
sitePrefixPath: "/global/"
URL to redirect to: "https://example.com/pics/../global/file.html"
```
I would like to say that the URL in question does not match their site prefix path because the string does not start with `"https://example.com/global/"`. If the URL was normalized it would match, but preferably it should be normalized before we receive it.

Of course, if they want to redirect to a relative path like `../global/file.html`, we handle the URL normalization properly. So the issue is only for redirects to absolute URLs that are not formatted correctly.

That said, I can take another look to see if we can conveniently normalize any URL string that we receive, including absolute ones.

Reply:
I changed UrlContext to now also normalize non-normalized URL inputs. It makes our test behavior identical to before.

I originally wanted to not handle these cases because it requires us to do more processing on the input we receive, and that can easily have weird bugs or side effects. (It helps if we can make _some_ assumptions about our input.) I think this normalization method is ok, though.